dhall-toml/src/Dhall/Toml.hs

```diff
@@ -25,68 +29,134 @@ import qualified Dhall.Parser
 import qualified Dhall.TypeCheck
 import qualified Toml.Type.TOML     as Toml.TOML
 import qualified Toml.Type.Value    as Toml.Value
+import qualified Toml.Type.AnyValue as Toml.AnyValue
 
 
 data CompileError
     = Unimplemented String
     | Unsupported (Expr Void Void)
+    -- tomland does not support records in multi-dimensional arrays, though it
+    -- is alloed by the spec
+    | UnsupportedArray (Expr Void Void)
     | NotARecord (Expr Void Void)
+    -- the latest TOML spec, v1.0.0 allows this but tomland has not
+    -- implemented it yet
+    -- NOTE: the only way to get this error is through enums
```

Comment:
The Dhall standard doesn't specify "enums". Do you mean unions or something else?